src/textual/command.py

```diff
@@ -1068,10 +1067,15 @@ def _select_or_command(
         if event is not None:
             event.stop()
         if self._list_visible:
+            command_list = self.query_one(CommandList)
             # ...so if nothing in the list is highlighted yet...
-            if self.query_one(CommandList).highlighted is None:
+            if command_list.highlighted is None:
                 # ...cause the first completion to be highlighted.
                 self._action_cursor_down()
+                # If there is one option, assume the user wants to select it
```

Comment:
Isn't this subject to the same issue as what we discussed earlier regarding having the first option always highlighted by default? If they can come in asynchronously, then when the user presses enter there may only be 1, but this doesn't mean there will always be one.

I think in VSCode and the browser URL bar, it also does things asynchronously, but it just selects the one that was at the top when the user presses enter.

Reply:
That does introduce a 100ms window where something could change. I'll see if I can remove that.

Reply:
I've removed the timer for now. In a future update, I'll implement the logic to highlight the first item.

Reply:
https://github.com/Textualize/textual/issues/4902